### DISCUSSION TITLE
List of sensors provided?

### DISCUSSION THREAD

**Discussion by calum-mcfarlane** [nathanmarlor/foxess_modbus#228] [General]
I was wondering if there is a list of the entities provided by this integration anywhere (or a particular bit of code I should look at)?

I'm looking to migrate from a (lightly tweaked) install of https://github.com/StealthChesnut/HA-FoxESS-Modbus, but as I have six months of history in an InfluxDB instance that I use for long term data history, I wanted to compare and see what name changes I'd need to deal with.

**Comment by canton7** [comment]
https://github.com/nathanmarlor/foxess_modbus/blob/main/custom_components/foxess_modbus/entities/entity_descriptions.py, look for ones which list your inverter type under addresses.

See also https://github.com/nathanmarlor/foxess_modbus/wiki/Migrating-from-HA-FoxESS-Modbus and #224

**Comment by calum-mcfarlane** [comment]
Thanks very much. It was partly #224 that made me think of asking. I'll do the compare and post it here for others' reference.

**Reply by canton7** [reply]
We should be almost 1:1 for the H1 (although we don't provide all of the template sensors which StealthChesnut recommended). For the H3, StealthChesnut used a different naming scheme, and we deviate from this in some places to keep the naming in line with the H1

**Reply by calum-mcfarlane** [reply]
I've got an H1 so that makes it straighforward from my POV. Presumably there's no harm in adding back any missing template sensors afterwards if needed (editing as required)?

**Reply by canton7** [reply]
That should be fine yep!

**Comment by Trickycs** [comment]
I'm still a little confused over the sensor definition and what's calculated / reset on a daily basis. I'm on an H1 inverter.
For example Feed-in total, says 11.2Kw, is that the daily total reset from midnight ? Grid consumption total 5.38Kw, same question. I have PV power total which is at 33Kwh but that for sure isn't today...
Is there any documentation that defines each of these ? I'm using a LAN cable to integrate which broadly seems to work if that makes any difference.

**Reply by canton7** [reply]
If you're connected with a LAN cable, the "Total" sensors are integration sensors -- home assistant sums up all of the readings from the corresponding power sensor to get a total, so it's the total since you started using the integration. While that might seem a bit useless, it's useful for things like the energy dashboard which require a "total" value.

If you're connected over RS485 we're able to fetch actual totals from the inverter for many values (but not PV1/PV2)

**Reply by Trickycs** [reply]
I have the novelty of one of the RS485 widgets arriving at the end of the month, so will entertain myself redoing the integration.
Is there a way to get things like "Daily Total" from the PV Power total, is HA able to "reset" to 0 at a particular point, like on the change of a calendar day ?
It's like that "Energy Distribution Today" chart which shows my solar today at 14.2 which seems about right as compared to the 33kwh which is the counter total.

**Reply by canton7** [reply]
Yep, they're called "Utility Meters" and you can create them from the "Helpers" tab under "Settings -> Devices & Services". https://www.home-assistant.io/integrations/utility_meter/. Use the "Total" sensor as the source
